compiler/ilgen/OMRVirtualMachineRegister.hpp

```diff
@@ -161,7 +161,8 @@ class VirtualMachineRegister : public TR::VirtualMachineState
    virtual void Adjust(TR::IlBuilder *b, TR::IlValue *amount)
       {
       TR::IlValue *off=b->Mul(amount,
-                       b->    ConstInteger(_integerTypeForAdjustments, _adjustByStep));
+                       b->    ConstInteger(b->typeDictionary()->PrimitiveType(amount->getDataType()),
```

Comment:
This is more dangerous, I think, because it requires the client to set up the type properly for every call to this Adjust service.

Reply:
Probably best to add a `ConvertTo(_integerTypeForAdjustments, amount)` here too.